query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/QueryRequestEligibilityValidator.java

```diff
@@ -0,0 +1,145 @@
+          .filter(Predicate.not(QueryRequestUtil::isDateTimeFunction))
+          .anyMatch(Predicate.not(QueryRequestUtil::isSimpleColumnExpression))) {
+        return QueryCost.UNSUPPORTED;
+      }
+
+      Set<String> referencedColumns = executionContext.getReferencedColumns();
+      Preconditions.checkArgument(!referencedColumns.isEmpty());
+      // all the columns in the request should have a mapping in the config
+      for (String referencedColumn : referencedColumns) {
+        if (prometheusViewDefinition.getPhysicalColumnNameForLogicalColumnName(referencedColumn)
+                == null
+            && prometheusViewDefinition.getMetricConfigForLogicalMetricName(referencedColumn)
+                == null) {
+          return QueryCost.UNSUPPORTED;
+        }
+      }
+
+      if (areAggregationsNotSupported(queryRequest.getAggregationList())) {
+        return QueryCost.UNSUPPORTED;
+      }
+
+      // if selection and groupBy should be on same column or simple attribute
+      if (selectionAndGroupByOnDifferentColumn(
+          queryRequest.getSelectionList(), queryRequest.getGroupByList())) {
+        return QueryCost.UNSUPPORTED;
+      }
+
+      if (isFilterNotSupported(queryRequest.getFilter())) {
+        return QueryCost.UNSUPPORTED;
+      }
+    } catch (Exception e) {
+      return QueryCost.UNSUPPORTED;
+    }
+    // value 1.0 so that prometheus is preferred over others
+    return new QueryCost(1.0);
+  }
+
+  private boolean selectionAndGroupByOnDifferentColumn(
+      List<Expression> selectionList, List<Expression> groupByList) {
+
+    Set<String> selections =
+        selectionList.stream()
+            .map(QueryRequestUtil::getLogicalColumnNameForSimpleColumnExpression)
+            .collect(Collectors.toSet());
+
+    Set<String> groupBys =
+        groupByList.stream()
+            .filter(Predicate.not(QueryRequestUtil::isDateTimeFunction))
+            .map(QueryRequestUtil::getLogicalColumnNameForSimpleColumnExpression)
+            .collect(Collectors.toSet());
+    return !selections.equals(groupBys);
+  }
+
+  private boolean areAggregationsNotSupported(List<Expression> aggregationList) {
+    return aggregationList.stream()
+        .filter(Predicate.not(QueryRequestUtil::isDateTimeFunction))
+        .anyMatch(
+            expression -> {
+              Function function = expression.getFunction();
+              if (function.getArgumentsCount() > 1) {
+                return true;
+              }
+              Expression functionArgument = function.getArgumentsList().get(0);
+              String attributeId = getLogicalColumnNameForSimpleColumnExpression(functionArgument);
+              if (!PrometheusFunctionConverter.supportedFunctions.contains(
+                  function.getFunctionName())) {
+                return true;
+              }
+              return prometheusViewDefinition.getMetricConfigForLogicalMetricName(attributeId)
+                  == null;
+            });
+  }
+
```

Comment:
Lets have a comment here that currently only `AND` filters supported. Later this can be extended for OR filter on same column (attributeId) expression.